esp/scm/ws_codesign.ecm

```diff
@@ -0,0 +1,44 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2019 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+ESPenum SigningMethodType : string
+{
+    gpg ("gpg")
+};
+
+ESPrequest SignRequest
+{
+    ESPenum SigningMethodType SigningMethod("gpg");
+    [cols(20)] string KeyIdentifier;
+    [password, cols(20)] string KeyPass;
+    [cols(70), rows(20)] string Text;
+};
+
+ESPresponse [exceptions_inline] SignResponse
+{
+    int RetCode;
+    string ErrMsg;
+    string SignedText;
+};
+
+ESPservice [auth_feature("WsCodeSignAccess:ACCESS"), version("1.0"), default_client_version("1.0"), exceptions_inline("./smc_xslt/exceptions.xslt")] ws_codesign
```

Comment:
If Russ changes the name as I suggested in my review of his PR then you will need to change these lines, I think?